components/consentRequestForm/index.jsx

```diff
@@ -206,6 +213,14 @@ export default function ConsentRequestForm({ agentDetails, agentWebId }) {
     setSelectedDate(date ? date.toISOString() : null);
   };
 
+  useEffect(() => {
+    if (!consentRequest) return;
+    setRequestedAccesses(getRequestedAccesses(consentRequest));
+    setExpirationDate(getExpiryDate(consentRequest));
+    setPurposes(getPurposeUrls(consentRequest));
+    setResourceOwnerWebId(getDataSubjectWebId(consentRequest));
+  }, [consentRequest]);
```

Comment:
This doesn't need to be a `useEffect` I don't think — I think you can just use initial state:

```
  const [requestedAccesses, setRequestedAccesses] = useState(() => getRequestedAccesses(consentRequest));
  const [expirationDate, setExpirationDate] = useState(() => getExpiryDate(consentRequest));
  const [purposes, setPurposes] = useState(() => getPurposeUrls(consentRequest));
  const [resourceOwnerWebId, setResourceOwnerWebId] = useState(() => getDataSubjectWebId(consentRequest));
```

And just have those methods return an empty value or null if there isn't a `consentRequest` (though there always should be one, as the prop is marked as required)

Reply:
That is, if the `consentRequest` changes the entire component will be re-rendered from scratch anyway, as the prop changed.